meson.build

```diff
@@ -108,6 +108,9 @@ warning_flags = [
   '-Wvla',
   '-Wwrite-strings'
 ]
+if not meson.is_cross_build()
```

Comment:
Can we detect mingw specifically instead?

Reply:
I don't think so, as this happens before `cc = meson.get_compiler('c')`

Reply:
How about using
```
if host_machine.system() == 'windows'
```

Reply:
Lets try that; fixed

Reply:
Maybe leave a comment with your GCC bug so future selves can find it when it's fixed to take this out.

Reply:
It might just be the mingw toolchain; investigating after I've done all my meetings for the day.